### PR TITLE
- Added some error handling to deal with broken systems

### PR DESCRIPTION
**Pull request by davekempe**
- Updated requirement to golang 1.8
- Made gometalinter.v1 optional

## Tips:

* commit message titles must be in the form:
```topic: Capitalized message with no trailing period```
or:
```topic, topic2: Capitalized message with no trailing period```

* golang code must be formatted according to the standard, please run:
```
make gofmt		# formats the entire project correctly
```
or format a single golang file correctly:
```
gofmt -w yourcode.go
```

* please rebase your patch against current git master:
```
git checkout master
git pull origin master
git checkout your-feature
git rebase master
git push your-remote your-feature
hub pull-request	# or submit with the github web ui
```

* after a patch review, please ping @purpleidea so we know to re-review:
```
# make changes based on reviews...
git add -p		# add new changes
git commit --amend	# combine with existing commit
git push your-remote your-feature -f
# now ping @purpleidea in the github PR since it doesn't notify us automatically
```

### CODE REVIEW

**Comment by davekempe** [comment]
OMG James

**Comment by davekempe** [comment]
Updated patch with more correct comments.

**Comment by davekempe** [comment]
Hey I think this is OK now @purpleidea

File: misc/make-deps.sh [hunk 1]
```diff
@@ -28,45 +33,45 @@ if [ -z "$YUM" -a -z "$APT" -a -z "$BREW" -a -z "$PACMAN" ]; then
...
 		# some go dependencies are stored in mercurial
-		$sudo_command $YUM install -y golang golang-googlecode-tools-stringer hg
+		$sudo_command $YUM install -y golang golang-googlecode-tools-stringer hg || errorhandler
 
 	fi
 	if [ ! -z "$APT" ]; then
 		$sudo_command $APT update
-		$sudo_command $APT install -y golang make gcc packagekit mercurial
+		$sudo_command $APT install -y golang make gcc packagekit mercurial  || errorhandler
 		# one of these two golang tools packages should work on debian
```
Comment (purpleidea) [inline misc/make-deps.sh]: I think you'll want to chain all three of these in a row as per the comment...

File: misc/make-deps.sh [hunk 2]
```diff
@@ -80,5 +85,11 @@ if [[ $ret != 0 ]]; then
 fi
 go get golang.org/x/tools/cmd/stringer	# for automatic stringer-ing
 go get github.com/golang/lint/golint	# for `golint`-ing
-go get -u gopkg.in/alecthomas/gometalinter.v1 && mv "$(dirname $(which gometalinter.v1))/gometalinter.v1" "$(dirname $(which gometalinter.v1))/gometalinter" && gometalinter --install	# bonus
+
+#check if its in the path if its set first
```
Comment (purpleidea) [inline misc/make-deps.sh]: this comment is ??? can you clarify please. also an its -> it's please

File: misc/make-deps.sh [hunk 3]
```diff
@@ -80,5 +85,11 @@ if [[ $ret != 0 ]]; then
...
+if [ -z $GOPATH ]; then
+	echo "GOPATH isnt set, you probably should fix that!"
```
Comment (purpleidea) [inline misc/make-deps.sh]: `$GOPATH isn't set, you should probably fix that`

File: misc/make-deps.sh [hunk 4]
```diff
@@ -80,5 +85,11 @@ if [[ $ret != 0 ]]; then
 fi
 go get golang.org/x/tools/cmd/stringer	# for automatic stringer-ing
 go get github.com/golang/lint/golint	# for `golint`-ing
-go get -u gopkg.in/alecthomas/gometalinter.v1 && mv "$(dirname $(which gometalinter.v1))/gometalinter.v1" "$(dirname $(which gometalinter.v1))/gometalinter" && gometalinter --install	# bonus
+
+#check if its in the path if its set first
+if [ -z $GOPATH ]; then
+	echo "GOPATH isnt set, you probably should fix that!"	
+	go get -u gopkg.in/alecthomas/gometalinter.v1 && mv ~/go/bin/gometalinter.v1 ~/go/bin/gometalinter && ~/go/bin/gometalinter --install	
+	else go get -u gopkg.in/alecthomas/gometalinter.v1 && mv "$GOPATH/bin/gometalinter.v1" "$GOPATH/bin/gometalinter" && $GOPATH/bin/gometalinter --install
```
Comment (purpleidea) [inline misc/make-deps.sh]: please put the else by itself and use ${GOPATH} instead.

File: misc/make-deps.sh [hunk 5]
```diff
@@ -4,6 +4,11 @@ XPWD=`pwd`
 ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd )"	# dir!
 cd "${ROOT}" >/dev/null
 
+function errorhandler () {
+	echo "Something went wrong! Hey, on the bright side if it breaks, you can keep both halves!"
```
Comment (purpleidea) [inline misc/make-deps.sh]: As much as I love this message, can you write something more specific to what happened, possibly mentioning a dep script? You could even do

```install thing || errorhandler "some message"

and use `"$1"` in the error handler for more info :)

**Comment by purpleidea** [comment]
PS: I forgot to mention it earlier, but you can use `fail_test` from test/util.sh

https://github.com/purpleidea/mgmt/blob/master/test/util.sh#L9

Here's an example of it being imported:

https://github.com/purpleidea/mgmt/blob/master/test/test-gometalinter.sh#L5

**Comment by purpleidea** [comment]
I'm going to close this due to inactivity, and also it has rotted against git master.

We've had a bunch of improvements since then, but if there are parts you'd still like to add (especially the gometalinter part) please open a new PR :)

Thank you again for your contribution, and hopefully you can try out the "new user" experience on a clean machine and tell us if you still have any issues.

Thanks!

**Pull request closed without merging**